tests/test_today_i_did_bot.py

```diff
@@ -8,19 +8,19 @@
 MOCK_TEST_FILE = '.testdata_sessions'
 
 
-def test_session(mocker):
-    mocked_channel_message = mocker.patch.object(TodayIDidBot, 'send_channel_message')
-    mocker.patch.object(TodayIDidBot, 'was_directed_at_me', return_value=True)
-    mocker.patch.object(TodayIDidBot, 'user_name_from_id', return_value=MOCK_PERSON)
-    mocker.patch.object(TodayIDidBot, 'connected_user', return_value='TodayIDidBot')
+@pytest.fixture
+def bot():
+    return TodayIDidBot('', rollbar_token='', elm_repo=None)
 
-    bot = TodayIDidBot('', rollbar_token='', elm_repo=None)
-    bot._last_sender = MOCK_PERSON
 
-    bot.parse_direct_message({
-        'user': MOCK_PERSON,
-        'channel': MOCK_CHANNEL,
-        'text': MOCK_START_SESSION_TEXT
+def test_start_session(mocker, bot, message_context):
```

Comment:
💭  this is magic, but maybe it's good magic. Test frameworks can get away with being a bit magic